File: Java-Core/src/main/java/prompto/error/InvalidDataError.java
package prompto.error;

import prompto.expression.IExpression;
import prompto.literal.TextLiteral;
import prompto.runtime.Context;

public class InvalidDataError extends ExecutionError {

	private static final long serialVersionUID = 1L;

	public InvalidDataError(String message) {
		super(message);
	}
	
	@Override
	public IExpression getExpression(Context context) {
		return new TextLiteral(super.getMessage());
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/error/InvalidDataError.js
var ExecutionError = require("./ExecutionError").ExecutionError;
var TextLiteral = require("../literal/TextLiteral").TextLiteral;

function InvalidDataError(message) {
	ExecutionError.call(this, message);
	return this;
}

InvalidDataError.prototype = Object.create(ExecutionError.prototype);
InvalidDataError.prototype.constructor = InvalidDataError;

InvalidDataError.prototype.getExpression = function(context) {
	return new TextLiteral("'" + this.message + "'");
};

exports.InvalidDataError = InvalidDataError;
